### PR TITLE
Remove unnecessary version code checks

### PR DESCRIPTION
**Pull request by RSBat**
## Purpose / Description
Since minSdkVersion was bumped to 16 (Jelly Bean) all these checks are
now constant and can be simplified

## Fixes
No issue

## Approach
Initially found one of them as lint issue, the rest was found with grep for `SDK_INT` and `getSdkVersion`

## How Has This Been Tested?

Since this is a purely mechanical change I just checked that it launches and basic scenarios work

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/sync/Tls12SocketFactory.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabsHelper.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574873&bid=18276683)

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java [hunk 1]
```diff
@@ -886,9 +886,7 @@ private void removeUnnecessaryAdvancedPrefs(PreferenceScreen screen) {
             CheckBoxPreference fixHebrewText = (CheckBoxPreference) screen.findPreference("fixHebrewText");
             CompatHelper.removeHiddenPreferences(this.getApplicationContext());
 
-            if (CompatHelper.getSdkVersion() >= 16) {
-                workarounds.removePreference(fixHebrewText);
-            }
+            workarounds.removePreference(fixHebrewText);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:889]: Can we remove this preference entirely?

Comment (RSBat) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:889]: Yes, it can be done. Should I remove related strings (like description) too?

**Comment by RSBat** [comment]
This should fix builds failing because of tests.
Learned that SDK_INT will be set to 0 in unit tests unless you use Robolectric. So maybe there are more incorrect tests

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabsHelper.java  -1
+ AnkiDroid/src/main/java/com/ichi2/libanki/sync/Tls12SocketFactory.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574873&bid=18276683)

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:889]: Yep! No point in burdening our translators with strings that people will never see.

File: AnkiDroid/src/test/java/com/ichi2/compat/customtabs/CustomTabActivityHelperTest.java [hunk 2]
```diff
@@ -58,7 +68,13 @@ public void invalidClientMeansFallbackIsCalled() {
         customTabActivityHelper.onServiceConnected(badClient);
 
         CustomTabActivityHelper.CustomTabFallback fallback = mock(CustomTabActivityHelper.CustomTabFallback.class);
-        CustomTabActivityHelper.openCustomTab(null, null, null, fallback);
+        Activity activity = mock(Activity.class);
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/compat/customtabs/CustomTabActivityHelperTest.java:71]: Sorry about the mess, thank you!

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java [hunk 3]
```diff
@@ -96,8 +96,6 @@ public static boolean hasScrollKeys() {
 
     public static void removeHiddenPreferences(Context context) {
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
-        if (getSdkVersion() >= 16) {
-            preferences.edit().remove("fixHebrewText").apply();
-        }
+        preferences.edit().remove("fixHebrewText").apply();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java:99]: should get rid of this at the same time I think since it is going away completely

File: AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabActivityHelper.java [hunk 4]
```diff
@@ -51,7 +52,7 @@
      * @param uri the Uri to be opened.
      * @param fallback a CustomTabFallback to be used if Custom Tabs is not available.
      */
-    public static void openCustomTab(Activity activity,
+    public static void openCustomTab(@NonNull Activity activity,
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabActivityHelper.java:55]: This is unrelated to the PR. I'm not against the change but PRs in general should stay focused on a single purpose

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java (hunk 3 above)
Comment (RSBat) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java]: done

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java (hunk 1 above)
Comment (RSBat) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java]: done

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java  -2
+ AnkiDroid/src/main/java/com/ichi2/libanki/sync/Tls12SocketFactory.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabsHelper.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574873&bid=18276683)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/sync/Tls12SocketFactory.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabsHelper.java  -1
+ AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574873&bid=18276683)

**Comment by mikehardy** [comment]
Something really strange happened in that second commit - it looks like it went back in git-time and now there are 102 files changed and a massive diff?

**Comment by RSBat** [comment]
Huge diff is caused by deleted strings. Or I should have deleted just the English version?
And what do you mean by "went back in git-time"?

**Comment by mikehardy** [comment]
Ah yes that explains it. I thought somehow you had pulled some local state in that was from translations that didn't exist yet, I hadn't thought that you would go in to other translations and change them. The general rule for translations is that we change things in 'values/' via PRs etc, all the 'values-XX/' directories are owned by crowdin.com and are changed after a round-trip to their translation server, which I handle

So definitely revert the changes to values-XX and just change the ones in values/

**Comment by RSBat** [comment]
Ok, I'll keep that in mind :smiley:

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/sync/Tls12SocketFactory.java  -1
+ AnkiDroid/src/main/java/com/ichi2/compat/customtabs/CustomTabsHelper.java  -1
+ AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574873&bid=18276683)

File: AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java [hunk 5]
```diff
@@ -53,9 +53,6 @@ private Hooks(Context context) {
         new Leech().installHook(this);
 
         // Preferences activated hooks
-        if (prefs.getBoolean("fixHebrewText", false)) {
-            HebrewFixFilter.install(this);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java:57]: Just a note for @Arthur-Milchior if you are working on Hooks that the HebreFixFilter hook is going away in this PR, it is dropped as no longer needed now that we are minSdkVersion of API16 for Android from #5665

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java:57]: I'm working on top of master, so if this disappear quickly, I would probably notice it soon. I'll thus consider this hook last.

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java:57]: @Arthur-Milchior merged just now - no more HebrewFixFilter on master

**Comment by mikehardy** [comment]
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users (the users' way of saying "Thanks!") to support maintenance and development of AnkiDroid.

We are looking through PRs merged in releases going back to release 2.10 (where we have good record keeping of what's in releases) to let our contributors know.

If you'd like to submit an expense related to your work here, to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!
